src/dns.ts

```diff
@@ -44,12 +46,16 @@ export class DNS implements DNSInterface {
    * Any new responses will be added to the cache for subsequent requests.
    */
   async query (domain: string, options: QueryOptions = {}): Promise<DNSResponse> {
-    const types = getTypes(options.types)
+    options.useRecordTypeValue = options.useRecordTypeValue ?? this.useRecordTypeValue
+    const types = getTypes(options.types, options.useRecordTypeValue)
     const cached = options.cached !== false ? this.cache.get(domain, types) : undefined
```

Comment:
see https://github.com/multiformats/js-dns/pull/5#discussion_r1532901944

Maybe:
```suggestion
    let cached
    if (options.cached !== false) {
      cached = this.cache.get(domain, types) ?? this.cache.get(domain, getTypes(types, !options.useRecordTypeValue))
    }
```